### PR TITLE
fixing max parallelism

### PR DESCRIPTION
**Pull request by hamersaw**
# TL;DR
Removing the `BarrierTick` code from the `TaskHandler` resulted in scenarios where parallelism is not incremented when tasks are frequently evaluated with no change in phase / phase version. This can result in more task executing in parallel than configured.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/3883

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by eapolinario** [comment]
is there any way we can verify this behavior with a local test (or a relatively cheap automated test)?

**Review by wild-endeavor — APPROVED** [review]

**Review by eapolinario — DISMISSED** [review]

**Review by eapolinario — APPROVED** [review]
